airflow/dags/monitor/test_retrieval.py

```diff
@@ -0,0 +1,291 @@
+from __future__ import annotations
+
+import json
+import logging
+import os
+from datetime import datetime
+from pathlib import Path
+from textwrap import dedent
+
+import pandas as pd
+from include.tasks.extract.utils.retrieval_tests import (
+    generate_answer,
+    get_or_create_drive_folder,
+    weaviate_search,
+    weaviate_search_mqr,
+)
+from include.tasks.extract.utils.weaviate.ask_astro_weaviate_hook import AskAstroWeaviateHook
+
+from airflow.decorators import dag, task
+from airflow.exceptions import AirflowException
+from airflow.models.param import Param
+from airflow.providers.google.suite.hooks.drive import GoogleDriveHook
+from airflow.providers.google.suite.hooks.sheets import GSheetsHook
+
+ask_astro_env = os.environ.get("ASK_ASTRO_ENV", "dev")
+test_questions_sheet_id = os.environ.get("TEST_QUESTIONS_SHEET_ID")
+askastro_endpoint_url = os.environ.get("ASK_ASTRO_ENDPOINT_URL")
+langchain_org_id = os.environ.get("LANGCHAIN_ORG")
+langchain_project_id = os.environ.get(f"LANGCHAIN_PROJECT_ID_{ask_astro_env.upper()}", None)
+azure_endpoint = os.environ["AZURE_OPENAI_USEAST_PARAMS"]
+google_domain_id = os.environ["GOOGLE_DOMAIN_ID"]
+
+drive_folder = "ask_astro/tests_" + ask_astro_env
+
+_WEAVIATE_CONN_ID = f"weaviate_{ask_astro_env}"
+_DRIVE_CONN_ID = "google_cloud_drive"
+
+WEAVIATE_CLASS = os.environ.get("WEAVIATE_CLASS", "DocsDev")
+
+logger = logging.getLogger("airflow.task")
+
+test_question_template_path = "include/data/test_questions_template.csv"
+
+expected_doc_count = 36860
```

Comment:
@mpgreg can we also parameterize expected_doc_count

Reply:
@vatsrahul1001 I think its best to just remove that check for now.  Its not directly related to retrieval tests but should rather be in a general monitoring DAG.  Need to think about how best to set the value because it will be different between dev, prod and test.

Reply:
@sunank200 if you agree I will just remove the doc count test from this PR.